source/extensions/http/header_validators/envoy_default/header_validator.cc

```diff
@@ -446,10 +446,15 @@ HeaderValidator::validatePathHeaderCharacters(const HeaderString& value) {
   }
 
   if (is_valid && iter != end && *iter == '#') {
-    // Validate the fragment component of the URI
-    ++iter;
-    for (; iter != end && is_valid; ++iter) {
-      is_valid &= testCharInTable(::Envoy::Http::kUriQueryAndFragmentCharTable, *iter);
+    if (!config_.strip_fragment_from_path()) {
+      return {HeaderValueValidationResult::Action::Reject,
+              UhvResponseCodeDetail::get().FragmentInUrlPath};
+    } else {
```

Comment:
nit: remove the `else` to remove a level of nesting (since the `if` block returns unconditionally)